runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -93,31 +106,30 @@ GC_ArrayletObjectModel::getArrayletLayout(J9Class* clazz, UDATA numberOfElements
 	}
 
 	/* CMVC 135307 : when checking for InlineContiguous layout, perform subtraction as adding to dataSizeInBytes could trigger overflow. */
-	if ((largestDesirableSpine == UDATA_MAX) || (dataSizeInBytes <= (largestDesirableSpine - minimumSpineSizeAfterGrowing - contiguousIndexableHeaderSize()))) {
+	if ((largestDesirableSpine == UDATA_MAX)
+		|| (dataSizeInBytes <= (largestDesirableSpine - minimumSpineSizeAfterGrowing - contiguousIndexableHeaderSize()))) {
 		layout = InlineContiguous;
 		if (0 == numberOfElements) {
 			/* Zero sized NUA uses the discontiguous shape */
 			layout = Discontiguous;
 		}
 	} else {
-		UDATA arrayletLeafSize = _omrVM->_arrayletLeafSize;
-		UDATA lastArrayletBytes = dataSizeInBytes & (arrayletLeafSize - 1);
+		uintptr_t arrayletLeafSize = _omrVM->_arrayletLeafSize;
+		uintptr_t lastArrayletBytes = dataSizeInBytes & (arrayletLeafSize - 1);
 
-		if (lastArrayletBytes > 0) {
+		if (isVirtualLargeObjectHeapEnabled() && (0 < dataSizeInBytes)) {
```

Comment:
is ' && (0 < dataSizeInBytes)' really necessary - 0-sized array is set to discontigous in the preceding case even for offheap enabled?